### PR TITLE
Explicit linter rules

### PR DESCRIPTION
**Pull request by zerospiel**
* set explicit linter rules for revive and govet
* granular exclusions for govet/revive
* removed unused exclusions
* added several extra linters

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -89,10 +99,35 @@ linters:
     - whitespace
 
 linters-settings:
+  depguard:
+    rules:
+      # Name of a rule.
+      main:
+        # Packages that are not allowed where the value is a suggestion.
+        deny:
+          - pkg: "github.com/pkg/errors"
```
Comment (s3rj1k) [inline .golangci.yml:108]: Hehe, I like this rule :)

File: .golangci.yml [hunk 2]
```diff
@@ -109,37 +144,188 @@ linters-settings:
...
+  prealloc:
+    for-loops: true
   revive:
-    enable-all-rules: true
+    # https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md
     rules:
       - name: add-constant
         disabled: true
+      - name: argument-limit
+        arguments: [6]
```
Comment (s3rj1k) [inline .golangci.yml:204]: I would suggest removing this linter, I understand that most of the time there is no sane need for 6+ args, but what if they are needed, this config will just be changed to new max value

File: .golangci.yml [hunk 3]
```diff
@@ -109,37 +144,188 @@ linters-settings:
...
+          - "sayRepetitiveInsteadOfStutters"
+      - name: file-header
+        disabled: true
+      - name: flag-parameter
       - name: function-length
         disabled: true
-      - name: max-public-structs
-        disabled: true # the api/* pkgs have lots of structs
+      - name: function-result-limit
+        arguments: [3]
```
Comment (s3rj1k) [inline .golangci.yml:260]: same here as for argument-limit

File: .golangci.yml [hunk 4]
```diff
@@ -109,37 +144,188 @@ linters-settings:
...
+          - "^[a-z][a-z0-9]{0,}$"
+      - name: import-shadowing
+      - name: imports-blocklist
+        disabled: true
+      - name: increment-decrement
+      - name: indent-error-flow
       - name: line-length-limit
         disabled: true
+      - name: max-control-nesting
+        arguments: [3]
```
Comment (s3rj1k) [inline .golangci.yml:275]: limit on nested ifs?

File: .golangci.yml [hunk 5]
```diff
@@ -109,37 +144,188 @@ linters-settings:
...
+      - name: max-public-structs
```
Comment (s3rj1k) [inline .golangci.yml:276]: Also suggest to disable that, some public APIs will fail that linter

File: .golangci.yml (hunk 4 above)
Comment (zerospiel) [reply .golangci.yml:275]: defautl is 5, will use it

File: .golangci.yml (hunk 5 above)
Comment (zerospiel) [reply .golangci.yml:276]: public APIs are removed from this check

**Comment by s3rj1k** [comment]
LGTM

**Review by Kshatrix — APPROVED** [review]

**Pull request merged**